### ISSUE TITLE
rspec tests failing

### ISSUE THREAD

**Issue by egh** [Tahi-project/tahi#799]
I can't seem to get the rspec tests to work. Problem 1:

```
/home/egh/c/tahi/spec/rails_helper.rb:56:in `<top (required)>': uninitialized constant VCR (NameError)
```

Easily fixed by adding `require 'vcr'` to `spec/rails_helper.rb`

After that there are problems with capybara-webkit that will take me a little more time to diagnose.

Do you have any advice on this? Thank you!

**Comment by afomi** [comment]
hi @egh, sounds like an environment issue.  I'm not seeing that on 2 of the Mac machines I work on normally.

One thing to try is to run `bundle exec rspec spec`.  Bundle will execute rspec with the bundled requirements specifed in Gemfile.

**Comment by egh** [comment]
Thanks for the confirmation. I'll keep trying. I am running using `bundler exec`.

**Comment by afomi** [comment]
Ah, no luck with `bundle exec`?  

Are you on another flavor of Linux? Most devs here are using OSX, but we have other machines/envs around if we need to troubleshoot this further.

**Comment by egh** [comment]
This is on ubuntu. I'll explore a bit more. I just wanted to check that the rspec tests were intended to run. On your CI server all the tests I saw were teaspoon tests

https://semaphoreapp.com/tahi-project/tahi/branches/master/builds/324

**Comment by afomi** [comment]
Hi @egh, thanks for the info.  Semaphore currently runs on Ubuntu 12.04, to my knowledge.  I did see `bundle exec rspec spec` and `bundle exec rspec engines` in Semaphore, which run just before the Teaspoon specs.

**Comment by egh** [comment]
Thans, Ryan. I just saw the `--format=documentation` part and completely missed that those lines were rspec tests.

In any case, I think the issue is that `dotenv` does not load from `.env` in rspec tests, but I have been using foreman to run the tests, which *does* load from `.env`. Will explore more tomorrow.

Thanks for all your help.

**Comment by egh** [comment]
Thanks again, Ryan. Here is one of the issues I came across trying to run the tests against a local postgres database.

https://github.com/Tahi-project/tahi/pull/804

**Comment by egh** [comment]
I can get the rspec tests to run now, but I'm getting a lot of failures. (The master build on semaphore is broken right now https://semaphoreapp.com/tahi-project/tahi/branches/master, but I get more errors than that. Will continue exploring.

**Comment by egh** [comment]
rspec does seem to load from `.env` files. I'm not sure what my issue there was.

Can you share your database.yml for test with me? Are you testing with postgres or in memory sqlite? I am trying to test with postgres but I keep getting the same error:

```
$ bundle exec rspec spec --fail-fast --order=defined
...
Failures:

  1) Editing paper Removing an existing collaborator
     Failure/Error: sign_in_page.sign_in author
     Selenium::WebDriver::Error::JavascriptError:
       arguments[0] is undefined
     # [remote server] http://127.0.0.1:31337/users/sign_in line 68 > Function:1:in `anonymous'
     # [remote server] http://127.0.0.1:31337/users/sign_in:68:in `handleEvaluateEvent'
     # ./spec/support/pages/sign_in_page.rb:5:in `sign_in'
     # ./spec/features/add_collaborators_spec.rb:15:in `block (2 levels) in <top (required)>'
```

**Comment by mikem** [comment]
The error message is from Selenium, @egh. Do the model specs pass? If they do, that should at least eliminate the DB as the root cause.

Here's the `test` section from `database.yml`:

```yaml
test:
  adapter: postgresql
  encoding: unicode
  database: tahi_test
  pool: 5
  username: tahi
  password:
```

**Comment by egh** [comment]
Thanks, Mike. That config looks identical to mine (except I have `host`). Yes, the model tests pass.

**Comment by egh** [comment]
Everything passes except the selinium tests, which is what I'd expect. Selenium tests are always the most fickle

**Comment by egh** [comment]
Just realized that I am not running the event server, which I suppose could be part of the problem.

**Comment by egh** [comment]
Doesn't seem to make a difference.

**Comment by mikem** [comment]
Yeah, you should be able to run the specs without having to start an event server.

Could you post more of the errors you're experiencing?

**Comment by egh** [comment]
Thanks to @mikem for looking over my shoulder and solving this problem in 2 seconds. FF 35 and selenium 2.44.0 do not play well together. Fixed in pr #809
